riscv/decode.h

```diff
@@ -390,38 +387,84 @@ static inline bool is_overlaped(const int astart, const int asize,
 }
 
 #define VI_NARROW_CHECK_COMMON \
+  require_vector;\
   require(P.VU.vlmul <= 4); \
   require(P.VU.vsew * 2 <= P.VU.ELEN); \
-  require(insn.rs2() + P.VU.vlmul * 2 <= 32);
+  require((insn.rs2() & (P.VU.vlmul * 2 - 1)) == 0); \
+  require((insn.rd() & (P.VU.vlmul - 1)) == 0); \
+  if (insn.v_vm() == 0 && P.VU.vlmul > 1) \
+    require(insn.rd() != 0);
 
 #define VI_WIDE_CHECK_COMMON \
   require_vector;\
   require(P.VU.vlmul <= 4); \
   require(P.VU.vsew * 2 <= P.VU.ELEN); \
-  require(insn.rd() + P.VU.vlmul * 2 <= 32); \
+  require((insn.rd() & (P.VU.vlmul * 2 - 1)) == 0); \
   if (insn.v_vm() == 0) \
     require(insn.rd() != 0);
 
-#define VI_CHECK_VREG_OVERLAP(v1, v2) \
-  require(!is_overlaped(v1, P.VU.vlmul, v2, P.VU.vlmul));
+#define VI_CHECK_MSS(is_vs1) \
+  if (P.VU.vlmul > 1) { \
+    require(!is_overlaped(insn.rd(), 1, insn.rs2(), P.VU.vlmul)); \
```

Comment:
"overlapped" is misspelled

Reply:
1. The tail-zero option is implemented in the transition between tail-zero in spec and non-tail-zero in some implementation.  It is ambiguous to tell whether the tail-zero is on or off with just spike executable.  If the spec has clearly define  "dont' zero", I will move it back to configurable only in compilation phase.
2. mis-spelling will be fixed.

Reply:
Thanks.  In this section of the spec, the behavior is now defined to be "don't zero": https://github.com/riscv/riscv-v-spec/blob/master/v-spec.adoc#54-prestart-active-inactive-body-and-tail-element-definitions

Reply:
Since the behavior is now defined to be "don't zero", I vote that we also remove the compile-time switch `TAIL_ZEROING` and related logic altogether. In my opinion, this will improve the readability of `decode.h`. On the other hand, if our goal is to support earlier versions of the spec, then we should leave it in.

Reply:
I have consulted some software users and they could agree with the removal.  I will remove it.

Reply:
Hi all, sorry for commenting in an old issue, have you preserved the `TAIL_ZEROING` switch? I think you don't, but in any case, I am using specification `0.7.1`. Do you know in which **commit** you had zeroing to re-insert it in my fork?

Reply:
You may try to revert b9d9e1eb to restore the tail-zeroing feature in this PR.